integration-tests/tests/run-notebook.js

```diff
@@ -26,6 +26,7 @@ const testRunNotebookFn = _.flow(
   await click(page, clickable({ text: 'Create Notebook' }))
   await click(page, clickable({ textContains: notebookName }))
   await waitForNoSpinners(page)
+  await findElement(page, clickable({ text: 'Edit' }))
   await click(page, clickable({ text: 'Edit' }))
```

Comment:
Doesn't a click first wait for the element to appear?

Reply:
It does:
```javascript
const click = async (page, xpath, options) => {
  return (await page.waitForXPath(xpath, options)).click()
}
```
The `findElement` should be redundant.

Reply:
There is a blocking spinner while the notebook preview loads. It's possible that the spinner hasn't yet rendered when the `await waitForNoSpinners(page)` call happens and then the test clicks (or at least thinks it clicks) "Edit" while the spinner is there, which means the click gets ignored.

This might be a case for a micro-sleep in the test (including a comment describing the gap that the sleep is bridging).

Reply:
I think the need for a micro-sleep is an exception. Usually, a click would update state in a way that would cause the spinner to appear right away. In some cases, though, there's an intermediate state change... click causes state change which causes another state change which renders the spinner. It's that extra step (which I haven't 100% confirmed is the case here but, knowing that this is having to call out to calhoun to render a preview and display it in an iframe, I wouldn't be surprised) that I think could be creating the gap that a micro-sleep could bridge.

Reply:
Makes sense. Thanks.